weld/subsystem/src/main/java/org/jboss/as/weld/deployment/processors/ExternalBeanArchiveProcessor.java

```diff
@@ -162,6 +162,14 @@ public void deploy(DeploymentPhaseContext phaseContext) throws DeploymentUnitPro
                 return;
             }
             for (DependencySpec dep : module.getDependencies()) {
+                if (!(dep instanceof ModuleDependencySpec)) {
+                    continue;
+                }
+                if (deploymentUnits.stream().anyMatch(d -> (((ModuleDependencySpec) dep).getName()).endsWith(d.getName()))) {
```

Comment:
My only "maybe" suggestion is earlier in this method when looping through the DeploymentUnits, built up a Set<String> containing "deployment." + d.getName().  Then here just do a contains check against that set.